src/Cli/Microsoft.TemplateEngine.Cli/Commands/CliPathInfo.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli.Commands
+{
+    internal sealed class CliPathInfo : IPathInfo
+    {
+        public CliPathInfo(
+            ITemplateEngineHost host,
+            IEnvironment environment,
+            string? settingsLocation)
+        {
+            UserProfileDir = GetUserProfileDir(environment);
+            GlobalSettingsDir = GetGlobalSettingsDir(settingsLocation);
+            HostSettingsDir = GetDefaultHostSettingsDir(host, globalDir: GlobalSettingsDir);
+            HostVersionSettingsDir = GetDefaultHostVersionSettingsDir(host, globalDir: GlobalSettingsDir);
+        }
+
+        public string UserProfileDir { get; }
+
+        public string GlobalSettingsDir { get; }
+
+        public string HostSettingsDir { get; }
+
+        public string HostVersionSettingsDir { get; }
+
+        private static string GetUserProfileDir(IEnvironment environment) => environment.GetEnvironmentVariable(
+            RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? "USERPROFILE"
+                : "HOME")
+            ?? throw new NotSupportedException("HOME or USERPROFILE environment variable is not defined, the environment is not supported");
+
+        private static string GetGlobalSettingsDir(string? settingsLocation)
+        {
+            var definedSettingsLocation = string.IsNullOrEmpty(settingsLocation)
+                ? CliFolderPathCalculator.DotnetHomePath
```

Comment:
```suggestion
                ? Path.Combine(CliFolderPathCalculator.DotnetHomePath, ".templateengine")
```

Reply:
We don't need to append a directory to `settingsLocation` specified via `debug:custom-settings` arg.